tests/server.rs

```diff
@@ -23,6 +23,48 @@ fn read_preface_in_multiple_frames() {
 }
 
 #[test]
+fn server_builder_set_max_concurrent_streams() {
+    let _ = ::env_logger::init();
+    let (io, client) = mock::new();
+
+    let mut settings = frame::Settings::default();
+    settings.set_max_concurrent_streams(Some(1));
+
+    let client = client
+        .assert_server_handshake()
+        .unwrap()
+        .recv_custom_settings(settings)
+        .send_frame(
+            frames::headers(1)
+                .request("GET", "https://example.com/")
+                .eos(),
+        )
+        .recv_frame(frames::headers(1).response(200).eos())
```

Comment:
To test that the setting was applied internally, this could also test that we send back a RST_STREAM of STREAM_REFUSED when the mock exceeds our advertised limit.

To so:

- Send the first request with a data frame also, but not immediately.
- Send a second request.
- Except to receive a `frames::reset(3).refused()`.
- Finish up the first request.
- Test done.

Reply:
Done (3d29b3f)!